src/library/trackset/setlogfeature.cpp

```diff
@@ -360,17 +410,103 @@ void SetlogFeature::slotJoinWithPrevious() {
                 if (m_playlistDao.copyPlaylistTracks(
                             currentPlaylistId, previousPlaylistId)) {
-                    m_lastRightClickedIndex = constructChildModel(previousPlaylistId);
                     m_playlistDao.deletePlaylist(currentPlaylistId);
-                    reloadChildModel(previousPlaylistId); // For moving selection
-                    emit showTrackModel(m_pPlaylistTableModel);
-                    activatePlaylist(previousPlaylistId);
                 }
             }
         }
     }
 }
 
+void SetlogFeature::slotLockAllChildPlaylists() {
+    lockOrUnlockAllChildPlaylists(true);
+}
+
+void SetlogFeature::slotUnlockAllChildPlaylists() {
+    lockOrUnlockAllChildPlaylists(false);
+}
+
+void SetlogFeature::lockOrUnlockAllChildPlaylists(bool lock) {
+    if (!m_lastRightClickedIndex.isValid()) {
+        return;
+    }
+    if (lock) {
+        qWarning() << "lock all child playlists of" << m_lastRightClickedIndex.data().toString();
+    } else {
+        qWarning() << "unlock all child playlists of" << m_lastRightClickedIndex.data().toString();
+    }
+    TreeItem* item = static_cast<TreeItem*>(m_lastRightClickedIndex.internalPointer());
+    if (!item) {
+        return;
+    }
+    const QList<TreeItem*> yearChildren = item->children();
+    if (yearChildren.isEmpty()) {
+        return;
+    }
+
+    QSet<int> ids;
+    for (const auto& pChild : yearChildren) {
+        bool ok = false;
+        int childId = pChild->getData().toInt(&ok);
+        if (ok && childId != kInvalidPlaylistId) {
+            ids.insert(childId);
+        }
+    }
+    m_playlistDao.setPlaylistsLocked(ids, lock);
+}
+
+void SetlogFeature::slotDeleteAllUnlockedChildPlaylists() {
+    if (!m_lastRightClickedIndex.isValid()) {
+        return;
+    }
+    TreeItem* item = static_cast<TreeItem*>(m_lastRightClickedIndex.internalPointer());
+    if (!item) {
+        return;
+    }
+    const QList<TreeItem*> yearChildren = item->children();
+    if (yearChildren.isEmpty()) {
+        return;
+    }
+    QString year = m_lastRightClickedIndex.data().toString();
+
+    QMessageBox::StandardButton btn = QMessageBox::question(nullptr,
+            tr("Confirm Deletion"),
+            //: %1 is the year
+            //: <b> + </b> are used to make the text in between bold in the popup
+            //: <br> is a linebreak
+            tr("Do you really want to delete all unlocked playlist in <b>%1</b>?<br><br>")
+                    .arg(year),
+            QMessageBox::Yes | QMessageBox::No,
+            QMessageBox::No);
+    if (btn == QMessageBox::No) {
+        return;
+    }
+    // Double-check, this is a weighty decision
+    btn = QMessageBox::warning(nullptr,
+            tr("Confirm Deletion"),
+            //: %1 is the year
+            //: <b> + </b> are used to make the text in between bold in the popup
+            //: <br> is a linebreak
+            tr("Deleting all unlokced playlist in <b>%1</b>.<br><br>"
```

Comment:
```suggestion
            tr("Deleting all unlocked playlist in <b>%1</b>.<br><br>"
```

Reply:
The current state already works for me. 
I have played with it and can imagine to have this, to make it a bit more informative, and pretend that we have already started after the first "Yes"
1. Do you really want to delete all unlocked playlist from 2021? Yes/No 
2. Deleting now 23 playlists from 2021. Abort/Ok

Reply:
Yes, that makes sense. In order to get the count I had to shift the dialog after the id list was created.